src/CachePeer.cc

```diff
@@ -123,6 +125,230 @@ CachePeer::connectTimeout() const
+        {
+            bool doneopts = false;
+            if (options.htcp_oldsquid)
+            {
+                os << (doneopts ? ',' : '=') << "oldsquid";
+                doneopts = true;
+            }
+            if (options.htcp_no_clr)
+            {
+                os << (doneopts ? ',' : '=') << "no-clr";
+                doneopts = true;
+            }
+            if (options.htcp_no_purge_clr)
+            {
+                os << (doneopts ? ',' : '=') << "no-purge-clr";
+                doneopts = true;
+            }
+            if (options.htcp_only_clr)
+            {
+                os << (doneopts ? ',' : '=') << "only-clr";
+                // doneopts = true; // uncomment if more opts are added
+            }
+        }
+    }
+#endif
+
+    if (options.no_netdb_exchange)
+        os << " no-netdb-exchange";
+
+#if USE_DELAY_POOLS
+    if (options.no_delay)
+        os << " no-delay";
+#endif
+
+    if (login)
+        os << " login=" << login;
+
+    if (mcast.ttl > 0)
+        os << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        os << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        os << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+
+    if (digest_url)
+        os << " digest-url=" << digest_url;
+
+#endif
+
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    else if (connection_auth == 2)
+        os << " connection-auth=auto";
+
+    secure.dumpCfg(os, "tls-");
+    os << '\n';
+}
+
+void
+CachePeer::reportStatistics (std::ostream& os)
+{
+    os << "\n" << std::setw(11) << std::left << typeString() <<
+        ": " << n_addresses << "\n" <<
+       "Host       : " << host << '/' << http_port << '/' <<
+           icp.port << "\n";
+    os << "Flags      :";
+    dumpOptions(os);
+
+    char ntoabuf[MAX_IPSTRLEN];
+    for (int i = 0; i < n_addresses; ++i)
+        os << "Address[" << i << "] : " <<
+            addresses[i].toStr(ntoabuf, MAX_IPSTRLEN) << "\n";
+
+    os << "Status     : " << (neighborUp(this) ? "Up" : "Down") << "\n" <<
+        "FETCHES    : " << stats.fetches << "\n" <<
+        "OPEN CONNS : " << stats.conn_open << "\n" <<
+        "AVG RTT    : " << stats.rtt << " msec\n";
+
+    if (!options.no_query) {
+        if (stats.last_query > 0)
+            os << "LAST QUERY : " <<
+                (squid_curtime - stats.last_query) << " seconds ago\n";
+        else
+            os << "LAST QUERY : none sent\n";
```

Comment:
I recommend not reporting what has not happened. If nothing else, the proposed text could mislead the reader into thinking that Squid did send some queries, but Squid was not able to send the last few queries (i.e. the last query sending attempts have failed).

```suggestion
```